pkg/common/parameters.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	ParameterKeyType                 = "type"
+	ParameterKeyReplicationType      = "replication-type"
+	ParameterKeyDiskEncryptionKmsKey = "disk-encryption-kms-key"
+
+	replicationTypeNone = "none"
+)
+
+// DiskParameters contains normalized and defaulted disk parameters
+type DiskParameters struct {
+	// Values: pd-standard OR pd-ssd
+	// Default: pd-standard
+	DiskType string
+	// Values: "none", regional-pd
+	// Default: "none"
+	ReplicationType string
+	// Values: {string}
+	// Default: ""
+	DiskEncryptionKMSKey string
+}
+
+// ExtractAndDefaultParameters will take the relevant parameters from a map and
+// put them into a well defined struct making sure to default unspecified fields
+func ExtractAndDefaultParameters(parameters map[string]string) (DiskParameters, error) {
+	p := DiskParameters{
+		DiskType:             "pd-standard",       // Default
+		ReplicationType:      replicationTypeNone, // Default
+		DiskEncryptionKMSKey: "",                  // Default
+	}
+	for k, v := range parameters {
+		if k == "csiProvisionerSecretName" || k == "csiProvisionerSecretNamespace" {
+			// These are hardcoded secrets keys required to function but not needed by GCE PD
+			continue
+		}
+		switch strings.ToLower(k) {
+		case ParameterKeyType:
+			if v != "" {
+				p.DiskType = v
```

Comment:
do we want toLower() here?

Reply:
hmmmm for in-tree we `toLower()` replication type but literally none of the other parameters. It feels like it doesn't matter so much but can definitely be added without issue